### PR TITLE
Add better transformer support for RoFormer

### PR DESCRIPTION
**Pull request by manish-p-gupta**
# What does this PR do?
Add BetterTransformer support for RoFormer (part of [#20372](https://github.com/huggingface/transformers/issues/20372))

## Before submitting
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests?

### CODE REVIEW

**Comment by manish-p-gupta** [comment]
Tried testing the model using pytest. Got the below error:
image

**Comment by manish-p-gupta** [comment]
Resolved the issue. I overlooked the bias assignment in the original Layer. Test cases seems to be passing now.

image

**Comment by fxmarty** [comment]
Hi, looks all good to me! Could you just run `make style` for code quality?

**Comment by manish-p-gupta** [comment]
Checked. There were couple of formatting fixes.

File: optimum/bettertransformer/models/__init__.py [hunk 1]
```diff
@@ -49,6 +50,7 @@ class BetterTransformerManager:
         "mbart": ("MBartEncoderLayer", MBartEncoderLayerBetterTransformer),
         "rembert": ("RemBertLayer", BertLayerBetterTransformer),
         "roberta": ("RobertaLayer", BertLayerBetterTransformer),
+        "roformer": ("RoFormerLayer", RoFormerLayerBetterTransformer),
```
Comment (fxmarty) [inline optimum/bettertransformer/models/__init__.py:53]: Great! Just something I noticed: I think the code for `BertLayerBetterTransformer` and the RoFormerLayerBetterTransformer you added is the same. Could you try here with this suggestion to see if it's fine? In which case the RoFormerLayerBetterTransformer is not needed.
```suggestion
        "roformer": ("RoFormerLayer", BertLayerBetterTransformer),
```

Comment (manish-p-gupta) [reply optimum/bettertransformer/models/__init__.py]: Yeah. The implementation are similar. Replaced it and it works fine.

**Comment by fxmarty** [comment]
Hi @manish-p-gupta , it looks good. Just a note: could you add https://github.com/huggingface/transformers/blob/main/pyproject.toml to the top level, and run `make style` again? We missed a black config for the repo, and you likely have a different local one, hence the huge diff :)

**Comment by manish-p-gupta** [comment]
Hey @fxmarty ,
Seems like I already have a pyproject.toml at the top level. There aren't any changes to push after `make style`. Anything else I might be missing here?

**Comment by fxmarty** [comment]
Oh my bad, there's already one indeed. I wonder why your `make style` changed so many files. I'll look into it.

**Review by fxmarty — APPROVED** [review]
Looks good to me, thanks for the addition!

**Pull request merged**